Seneca-CDOT/telescope
# Fixes #2792 Set up Docusaurus as a Microservice

## Issue This PR Addresses

Fixes #2792
Fixes #2746 
Fixes #290 



## Type of Change




## Description



This sets up Docusaurus in a Docker container that runs locally on port 4631.

So far, I only have the Docusaurus website running in a Docker container.  To test the container:

1. Add my repository as a remote: `git remote add cindyledev https://github.com/cindyledev/telescope.git`
2. Fetch my branches: `git fetch cindyledev`
3. Checkout my branch: `git checkout docusaurus`
4. Change directory to /src/docs: `cd src/docs`
5. Build the Docker image: `docker build --target development -t docs:dev .`
6. Run the Docker container: `docker run -p 4631:4631 docs:dev`
7. Go to localhost:4631 to see the Docusaurus website.

TODO
Nginx stuff

<hr>

How to test this PR

1. Add my repository as a remote: `git remote add cindyledev https://github.com/cindyledev/telescope.git`
2. Fetch my branches: `git fetch cindyledev`
3. Checkout my branch: `git checkout docusaurus`
4. Install dependencies: `pnpm install`
5. Start the Docker containers: `pnpm services:start`

Go to localhost:4631 to see the Docusaurus website.

## Steps to test the PR



## Checklist
